workshop/5 - Security/README.md

```diff
@@ -0,0 +1,259 @@
+# 5 - Security
+
+Cloud applications are often composed of multiple Azure services. Communication between those services needs to be secure. Enforcing secure authentication, authorization, and accounting practices in your application is essential to your security posture. At this phase in the cloud journey, you should use managed identities, secrets management, and private endpoints. Here are the security recommendations for the reliable web app pattern.
+
+## Managed identities
+
+Managed identities provide an automatically managed identity in Azure Active Directory (Azure AD) that applications can use when they connect to resources that support Azure AD authentication.
+
+When you grant managed identities access to a resource, you should always grant the least permissions needed.
+
+Let's see how managed identities can help us remove connection strings from our code.
+
+1. Use the main sample at the ``azd-sample`` directory. If you modified the sample in the last lesson, undo the modifications.
+1. Open the [Azure portal](https://portal.azure.com) and browse for resource groups. Search the Resource Group for the RWA Sample.
+1. Open the App Service from the Web Service within the resource group.
+1. Click on the **Identity** menu item in the left navigation.
+1. From the **System assigned** tab, click **On** to enable the system assigned identity. Then click **Save**.
+
+    ![Screenshot of the App Service identity](../images/5-Security/app-service-identity.png)
+
+1. Click yes at the prompt to enable the identity.
+1. Make note of what the web application's name is. It will probably be easiest to copy it to a text file for reference later.
+    _It will be named similar to: **groupname-web-mhl5v7j5rkxya**_
+1. Go back to the resource group and open the Web App - App Configuration resource.
+1. Click on the **Access control (IAM)** menu item in the left navigation.
+1. Click the **+ Add** button. Then click **Add role assignment**.
+1. Select **App Configuration Data Reader** and click **Next**.
+1. In the **Members** tab, change **Assign access to** to **Managed identity**.
+1. Click **+ Select members**.
+1. In the **Select managed identities** blade, from the **Managed identity** drop down, select **App Service** under the **System-assigned managed identity** section.
+
+    ![Screenshot of picking the managed identity](../images/5-Security/selecting-managed-identity.png)
+
+1. Enter your web application's name in the **Select** field and then choose it. It should not appear under the **Selected members** section. Then click **Select**.
+1. The blade will disappear and you should see your web application's name under the **Members** section. Click **Review and assign**.
+1. Go through the same steps as above, but this time add your user account as a **App Configuration Data Reader**. _Note: you will use **User, group, or service principal** instead of **Managed identity** to search for your username to add._
```

Comment:
note: you must wait until the user is fully added to an access policy before moving on to the next, otherwise it will fail.